### PR TITLE
Exercise 3 markups

### PR DESCRIPTION
**Pull request by PerryJones901**
The markups for Ex 3:

- Made TrelloConfig, TrelloHelper and ApiHelper classes
- Improved Int and E2E tests

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -15,7 +15,6 @@ RUN poetry install
 
 FROM base AS production
 # COPY across Config
-COPY env_vars.py env_vars.py
```
Comment (Jonesey13) [inline Dockerfile:18]: Definitely not exercise 3 :)

File: api/trello_helper.py [hunk 2]
```diff
@@ -1,54 +1,58 @@
...
-    url = f"{env.BASE_URL}/boards/{id}"
-    return api.delete(url)
\ No newline at end of file
+class TrelloHelper:
+    def __init__(self, trello_config: TrelloConfig):
+        self.config = trello_config
+        self.api = ApiHelper(trello_config)
+        
+    def build_board_url(self,uri) -> str:
+        return f"{self.config.BASE_URL}/boards/{self.config.BOARD_ID}{uri}"
```
Comment (Jonesey13) [inline api/trello_helper.py]: A very minor point but I'd be tempted to put a "/" before the uri here

File: .gitignore [hunk 3]
```diff
@@ -124,3 +124,6 @@ geckodriver.exe
 
 ## Gunicorn files ##
 gunicorn.log
+
+## VS Code ##
+.vscode/
```
Comment (Jonesey13) [inline .gitignore]: I'd also add `.venv` to this file.

File: tests/test_integration.py [hunk 4]
```diff
@@ -46,3 +49,11 @@ def mock_get_requests(*args, **kwargs):
...
     response = client.get('/')
     assert response.status_code == 200
+
+@mock.patch('requests.get', side_effect=mock_get_requests)
+def test_index_page_contains_card_names(mock_get_requests, client):
+    response_data = client.get('/').data
+    soup = BeautifulSoup(response_data, 'html.parser')
+    all_task_id_and_name_elements = soup.find_all(class_="task-id-and-name")
+    for i in range(3):
+        assert (const.CARD_ARR[i]["name"] in all_task_id_and_name_elements[i].string)
```
Comment (Jonesey13) [inline tests/test_integration.py]: You could use `enumerate` here:
```suggestion
    for index, card in enumerate(const.CARD_ARR):
        assert (card["name"] in all_task_id_and_name_elements[index].string)
```

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
The tests now run nicely but there still isn't instructions on how to run the tests in the `README.md` file.

Please see the PR comments for some other minor points.

File: Dockerfile (hunk 1 above)
Comment (PerryJones901) [reply Dockerfile:18]: That is true! Though `env_vars.py` is no longer needed by docker (and I've just realised it's not needed by anything else either! So I'll delete it).

**Review by hugh-emerson — APPROVED** [review]
Just looking at the latest bit - looks good! Merge when you're ready.

**Pull request merged**